web/src/Components/Optimization/OptimizationContainer.tsx

```diff
@@ -18,40 +18,77 @@ export interface ProductResult {
 export const OptimizationContainer = ({
   products,
   mode,
-  addCombination,
+  addCombinationsFromOptimization,
   value,
 }: OptimizationContainerProps): ReactElement => {
-  const [optimizationData, setOptimizationData] = useState<any>(undefined)
+  const [densityOptimizationData, setDensityOptimizationData] = useState<any>(undefined)
 
-  const handleUpdate = (optimizationData: any) => {
-    setOptimizationData(optimizationData)
+  const convertDensityOptimizationToSacks = (densityOptimizationData: OptimizationData): ProductsInCombination => {
+    let sackProducts: ProductsInCombination = {}
+    Object.values(densityOptimizationData.products).map(product => {
+      const SACK_KG: number = 25
+      const sackValue: number = Math.round((product.value * densityOptimizationData.chosenVolume) / SACK_KG)
+      if (sackValue > 0) {
+        sackProducts = { ...sackProducts, [product.id]: { id: product.id, value: sackValue, percentage: 0 } }
+      }
+    })
 
-    if (Object.keys(optimizationData.products).length === 0) {
+    let totalNumberOfSacks: number = 0
+    Object.values(sackProducts).map(product => {
+      totalNumberOfSacks += product.value
+    })
+
+    Object.values(sackProducts).map(product => {
+      sackProducts[product.id].percentage = (product.value / totalNumberOfSacks) * 100
+    })
+
+    //NOTE: since densities can be rounded down to 0 sacks, number of products in optimization for density and sacks can differ
+    return sackProducts
+  }
+
+  const handleUpdate = (densityOptimizationData: OptimizationData) => {
+    if (Object.keys(densityOptimizationData.products).length === 0) {
       alert('Could not find a solution. Try changing some parameters')
       return
     }
+    const sackOptimizationValues = convertDensityOptimizationToSacks(densityOptimizationData)
+    setDensityOptimizationData(densityOptimizationData)
+
     const datetime = new Date()
-    const name = `Optimized at ${datetime
-      .getHours()
-      .toString()
-      .padStart(2, '0')}:${datetime
+    const timeString = `${datetime.getHours().toString().padStart(2, '0')}:${datetime
       .getMinutes()
       .toString()
       .padStart(2, '0')}:${datetime.getSeconds().toString().padStart(2, '0')}`
+    const sackResultName = `Optimized for sacks at ${timeString}`
+    const densityResultName = `Optimized for densities at ${timeString}`
```

Comment:
This makes it sound like there are different optimizations for sacks and densities.
Another naming scheme could be "Optimized at {time} (sacks)"